contracts/DXswapFeeReceiver.sol

```diff
@@ -0,0 +1,141 @@
+pragma solidity =0.5.16;
+
+import './interfaces/IDXswapFactory.sol';
+import './interfaces/IDXswapPair.sol';
+import './interfaces/IWETH.sol';
+import './libraries/TransferHelper.sol';
+import './libraries/SafeMath.sol';
+
+
+contract DXswapFeeReceiver {
+    using SafeMath for uint;
+
+    address public owner;
+    mapping(address => address) public pairOwners;
+    IDXswapFactory public factory;
+    address public WETH;
+
+    constructor(address _owner, address _factory, address _WETH) public {
+        owner = _owner;
+        factory = IDXswapFactory(_factory);
+        WETH = _WETH;
+    }
+
+    function transferOwnership(address newOwner) external {
+        require(msg.sender == owner, 'DXswapFeeReceiver: FORBIDDEN');
+        owner = newOwner;
+    }
+    
+    // returns sorted token addresses, used to handle return values from pairs sorted in this order
+    function sortTokens(address tokenA, address tokenB) internal pure returns (address token0, address token1) {
+        require(tokenA != tokenB, 'DXswapFeeReceiver: IDENTICAL_ADDRESSES');
+        (token0, token1) = tokenA < tokenB ? (tokenA, tokenB) : (tokenB, tokenA);
+        require(token0 != address(0), 'DXswapFeeReceiver: ZERO_ADDRESS');
+    }
+
+    // calculates the CREATE2 address for a pair without making any external calls
+    // Taken from DXswapRouter, removed the factory parameter
+    function pairFor(address tokenA, address tokenB) internal view returns (address pair) {
+        (address token0, address token1) = sortTokens(tokenA, tokenB);
+        pair = address(uint(keccak256(abi.encodePacked(
+                hex'ff',
+                factory,
+                keccak256(abi.encodePacked(token0, token1)),
+                hex'2db943b381c6ef706828ea5e89f480bd449d4d3a2b98e6da97b30d0eb41fb6d6' // init code hash
+            ))));
+    }
+    
+    // fetches and sorts the reserves for a pair
+    function getReserves(address tokenA, address tokenB) internal view returns (uint reserveA, uint reserveB) {
+        (address token0,) = sortTokens(tokenA, tokenB);
```

Comment:
Couple questions about the left side of the equation.  Why is it `token0` if it is a pair?  And why is it in parenthesis?

Reply:
Oh, nevermind

Reply:
I guess is to save gas, avoid having a token1 varibale thta you dont need since you only need to know the first token when you sort it, and after you get the reserves (sorted by token address) you know which reserve goes to which token.